tests/integration/test_bayesian_optimization.py

```diff
@@ -164,3 +172,67 @@ def build_model(data: Dataset) -> GaussianProcessRegression:
         acquisition_function = acquisition_rule._acquisition_function
         if isinstance(acquisition_function, AcquisitionFunctionClass):
             assert acquisition_function.__call__._get_tracing_count() == 3  # type: ignore
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, acquisition_rule",
+    [
+        (5, DiscreteThompsonSampling(1000, 50)),
+    ],
+)
+def test_two_layer_dgp_optimizer_finds_minima_of_michalewicz_function(
+    num_steps: int, acquisition_rule: AcquisitionRule[TensorType, SearchSpace]
+) -> None:
+
+    search_space = Box([0, 0], [pi, pi])
```

Comment:
there should be michalewicz_search_space variable in objectives, right?

Reply:
Yep, done